src/test/unit/model/indexing/assign_cl_test.cpp

```diff
@@ -0,0 +1,413 @@
+#ifdef STAN_OPENCL
+#include <iostream>
+#include <stdexcept>
+#include <vector>
+#include <stan/model/indexing/assign.hpp>
+#include <stan/model/indexing/assign_cl.hpp>
+#include <stan/model/indexing/rvalue.hpp>
+#include <stan/model/indexing/rvalue_cl.hpp>
+#include <stan/math.hpp>
+#include <gtest/gtest.h>
+#include <test/unit/util.hpp>
+#include <test/unit/pretty_print_types.hpp>
+#include <tuple>
+
+using stan::model::assign;
+using stan::model::rvalue;
+
+using stan::model::index_max;
+using stan::model::index_min;
+using stan::model::index_min_max;
+using stan::model::index_multi;
+using stan::model::index_omni;
+using stan::model::index_uni;
+
+template <typename T>
+T opencl_index(T i) {
+  return i;
+}
+stan::math::matrix_cl<int> opencl_index(const index_multi& i) {
+  return stan::math::to_matrix_cl(i.ns_);
+}
+
+template <typename T, stan::require_not_rev_kernel_expression_t<T>* = nullptr>
+void set_adjoints1(T& v) {
```

Comment:
They are just set adjoints in two different patterns, so as not to make any tests pass just because all the adjoints are the same. What would you name them?

Reply:
Why not have one function that brings in what you set the adjoint to? Then you can just call it `set_adjoints(...)`

Reply:
That would be possible if all adjoints were set to the same value in one call of `set_adjoints`. However, to avoid missing any errors, all adjoints need to have different values, even ones set in the same call. So we have two functions, each setting adjoints according to different pattern. This way all adjoints are different.